### DISCUSSION TITLE
search with *

### DISCUSSION THREAD

**Discussion by paulobreim** [sepinf-inc/IPED#1395] [Q&A] [answered]
I was doing a search with the term "select" in all files that start with the letters scm and extension in PHP language. The objective was to check the programs that access a database using select.
So I put in the filter:
select && name:scm* && ext:php
The search returned ok, but included in the result, files that do not start with scm but have a _scm.
for example: xpto_scmpxxx.php
is this correct or is it a bug?
I also tested with:
select && name:scm* && ext:php && -"_scm"
In this case it no longer brought the files that had _scm in the name, but also eliminated the ones that had scm in the beginning and in the middle, for example: scm001_scm.php

pcb

**Comment by wladimirleite** [comment, marked as answer]
That is the expected behavior.
The underscore character is not index by default, so it is considered as a separator. 
You can change this behavior (i.e. treat underscore or other characters as a "regular character", like numbers and letters) in the configuration file "IndexTaskConfig.txt", entry "extraCharsToIndex".

This is related to discussion https://github.com/sepinf-inc/IPED/discussions/1236.

A more detailed explanation:
https://github.com/sepinf-inc/IPED/wiki/User-Manual#indexing

**Reply by paulobreim** [reply]
thank you so much. I didn't even remember that
